docs/debug.md

```diff
@@ -0,0 +1,196 @@
+# Debugging SOCI snapshotter
+
+This document outlines where to find/access logs and metrics for the snapshotter. It attempts to provide some common error paths that a user might run into while using the snapshotter and provides some guidance on how to either root-cause the issue or resolve it.
+
+- [Debugging SOCI snapshotter](#debugging-soci-snapshotter)
+- [Finding Logs / Metrics](#finding-logs--metrics)
+  - [Logs](#logs)
+  - [Metrics](#metrics)
+    - [Accessing Metrics](#accessing-metrics)
+    - [Metrics Emitted](#metrics-emitted)
+- [Common Scenarios](#common-scenarios)
+  - [`rpull`](#rpull)
+    - [No lazy-loading](#no-lazy-loading)
+    - [`rpull` Taking An Abnormal Amount Of Time](#rpull-taking-an-abnormal-amount-of-time)
+    - [Background Fetching](#background-fetching)
+  - [Running Container](#running-container)
+    - [FUSE Read Failures](#fuse-read-failures)
+- [Debugging Tools](#debugging-tools)
+  - [CLI](#cli)
+  - [CPU Profiling](#cpu-profiling)
```

Comment:
Was this TOC manually inserted? We use an automatic tool for TOCs in some other md docs.

Reply:
No this was generated via a vs-code plugin https://marketplace.visualstudio.com/items?itemName=shd101wyy.markdown-preview-enhanced